webpack/assets/javascripts/foreman_dashboard.js

```diff
@@ -0,0 +1,138 @@
+import $ from 'jquery';
+import { doesDocumentHasFocus } from './react_app/common/document';
+import { notify } from './foreman_toast_notifications';
+import { activateTooltips } from './foreman_tools';
+import { translate as __ } from './react_app/common/I18n';
+
+$(document).on('ContentLoad', () => {
+  if (window.foreman_url(window.location.pathname) === '/') {
+    startGridster();
+    autoRefresh();
+  }
+});
+
+let refreshTimeout;
+
+function autoRefresh() {
```

Comment:
I think this function can be simplified quite a lot, and not start a timer at all if it's not needed:
```es6
function autoRefresh() {
  clearTimeout(refreshTimeout);
  const autoRefreshIsOn = $('.auto-refresh').hasClass('on');
  if (doesDocumentHasFocus() && autoRefreshIsOn) {
    refreshTimeout = setTimeout(() => {
      window.Turbolinks.visit(window.location.href);
    }, 60000);
  }
}

Reply:
Your version does lose one benefit: if the window doesn't have focus at the end of the timeout then it's still reloaded. By starting the timer and checking if it needs a refresh before doing so it's slightly more efficient.

Reply:
@tbrisker I would prefer to refactor outside of this PR

Reply:
fair enough.

Reply:
@tbrisker thanks, would you mind merging then? :)